### PR TITLE
docs: add notebooks for modpath7 examples 3-4

### PR DESCRIPTION
**Pull request by w-bonelli**
Add notebooks for problems 3 and 4 from the examples bundled with [the MODPATH 7 distribution](https://www.usgs.gov/software/modpath-particle-tracking-model-modflow)
- [x] example 3
- [x] example 4

Both examples are reproduced programmatically.

Could later be updated to compare MODPATH and PRT (when released)

### CODE REVIEW

File: examples/data/mp7/ex04_mf6/ex04_mf6.ims [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  UNDER_RELAXATION  NONE
+  UNDER_RELAXATION_THETA    0.000000    
+  UNDER_RELAXATION_KAPPA    0.000000    
+  UNDER_RELAXATION_GAMMA    0.000000    
+  UNDER_RELAXATION_MOMENTUM    0.000000    
+END Nonlinear
+
+BEGIN LINEAR
+  INNER_MAXIMUM  100
+  INNER_HCLOSE   0.1000000E-05
```
Comment (jdhughes-usgs) [inline examples/data/mp7/ex04_mf6/ex04_mf6.ims:20]: Could you change `inner_hclose` to `inner_dvclose`

**Review by jdhughes-usgs — COMMENTED** [review]
Using `edgecolor` or `facecolor` will eliminte the warning in `flopy_modpath7_structured_transient.ipynb` which is causing the warning that is showing your file absolute path. Did we ever find a way to scrub this type of data?

If you reverse the order of your polygons in `flopy_modpath7_unstructured_transient.ipynb` will this eliminate the GeoJSON wayning?

**Comment by w-bonelli** [comment]
Thanks @jdhughes-usgs, fixed warnings.

Absolute paths were cleaned up in https://github.com/modflowpy/flopy/pull/1711, but it looks like paths aren't always computed correctly now (e.g. executable paths in notebooks here). Opened https://github.com/modflowpy/flopy/pull/1728 to address

**Comment by w-bonelli** [comment]
Fixed executable paths here as per approach proposed in 1728

**Review by langevin-usgs — COMMENTED** [review]
Looks great, @w-bonelli!  Few minor things for your consideration.

* "stress_period_data={0: dd, 1: dd, 2: dd}"  If the drains are the same for every stress period, you only need to specify them for the first one and MODFLOW will automatically keep applying them.
* The figure that uses the add_patches is a little confusing.  The legend has color fill symbols, but not polygon symbols.  And maybe features are overlying one another, but the I don't see all the legend colors on the plot.  Maybe a combination of symbols and filled polygons would be best?
* We often shrink the colorbar using shrink=0.5
* The unstructured notebook looks great.

**Comment by w-bonelli** [comment]
Fixed pathline colors in existing notebooks to address #1732, also updated to suppress/capture model logs and avoid print statements showing executable path (per #1728)

**Pull request merged**